KubeArmor/core/kubeUpdate.go

```diff
@@ -486,6 +476,29 @@ func (dm *KubeArmorDaemon) WatchK8sPods() {
 					pod.Labels[k] = v
 				}
 
+				if len(event.Object.ObjectMeta.OwnerReferences) > 0 {
```

Comment:
what happens if there are multiple event.Object.ObjectMeta.OwnerReferences .. currently we are using ref at index 0 .. I am just worried that we may not know which index holds the right ref that we may be interested in.

Reply:
We go through all OwnerReferences and use https://github.com/kubernetes/design-proposals-archive/blob/main/api-machinery/controller-ref.md#api to conclude whether a reference is to the pod's managing controller.
There can be only one controller reference per pod.